test/malli/transform_test.cljc

```diff
@@ -1125,41 +1125,79 @@
-               :equals1 "kikka"
-               :equals2 "kikka"
-               :equals3 "1"
-               :equals4 "1.1"}
-        expected {:enum1 :kikka
-                  :enum2 'kikka
-                  :enum3 1
-                  :enum4 1.1
-                  :equals1 :kikka
-                  :equals2 'kikka
-                  :equals3 1
-                  :equals4 1.1}]
-    (testing "decoding"
-      (testing "is not enabled by default"
-        (is (= value (m/decode schema value nil))))
-      (testing "works with json and string transformers"
-        (is (= expected (m/decode schema value (mt/json-transformer))))
-        (is (= expected (m/decode schema value (mt/string-transformer))))))
-    (testing "encoding"
-      (testing "is not enabled by default"
-        (is (= expected (m/encode schema expected nil))))
-      (testing "works with json and string transformers"
-        (is (= value (m/encode schema expected (mt/json-transformer))))
-        (is (= value (m/encode schema expected (mt/string-transformer))))))))
+(def child-inference-test-schema
+  [:map
+   [:enum1 [:enum :kikka :kukka]]
+   [:enum2 [:enum 'kikka 'kukka]]
+   [:enum3 [:enum 1 2]]
+   [:enum4 [:enum 1.1 2.2]]
+   [:equals1 [:= :kikka]]
+   [:equals2 [:= 'kikka]]
+   [:equals3 [:= 1]]
+   [:equals4 [:= 1.1]]])
+
+(deftest child-inference-default-test
+  (let [encoded {:enum1 "kikka"
+                 :enum2 "kikka"
+                 :enum3 "1"
+                 :enum4 "1.1"
+                 :equals1 "kikka"
+                 :equals2 "kikka"
+                 :equals3 "1"
+                 :equals4 "1.1"}
+        decoded {:enum1 :kikka
+                 :enum2 'kikka
+                 :enum3 1
+                 :enum4 1.1
+                 :equals1 :kikka
+                 :equals2 'kikka
+                 :equals3 1
+                 :equals4 1.1}]
+    (testing "decoding is not enabled by default"
+      (is (= encoded (m/decode child-inference-test-schema encoded nil))))
+    (testing "encoding is not enabled by default"
+      (is (= decoded (m/encode child-inference-test-schema decoded nil))))))
+
+(deftest child-inference-json-test
+  (let [encoded {:enum1 "kikka"
+                 :enum2 "kikka"
+                 :enum3 1
+                 :enum4 1.1
+                 :equals1 "kikka"
+                 :equals2 "kikka"
+                 :equals3 1
+                 :equals4 1.1}
+        decoded {:enum1 :kikka
+                 :enum2 'kikka
+                 :enum3 1
+                 :enum4 1.1
+                 :equals1 :kikka
+                 :equals2 'kikka
+                 :equals3 1
+                 :equals4 1.1}]
+    (testing "decoding children using the json transformer works"
+      (is (= decoded (m/decode child-inference-test-schema encoded (mt/json-transformer)))))
+    (testing "encoding children using the json transformer works"
+      (is (= encoded (m/encode child-inference-test-schema decoded (mt/json-transformer)))))))
```

Comment:
I'd like to see a case here for what happens when using the stringy `encoded` value with the json-transformer. The old test covered that.

Reply:
Added that test + threw in some extra cases that assert that encoding or decoding values that are already encoded or decoded doesn't do anything.